MoreLinq.Test/SortedMergeTest.cs

```diff
@@ -54,6 +54,17 @@ public void TestSortedMergeDisposesOnError()
             }
         }
 
+        /// <summary>
+        /// Verify that SortedMerge do not call MoveNext method eagerly
+        /// </summary>
+        [Test]
+        public void TestSortedMergeDoNotCallMoveNextEagerly()
+        {
+            var sequenceA = TestingSequence.Of(1, 3);
+            var sequenceB = MoreEnumerable.From(() => 2, () => throw new TestException());
+            sequenceA.SortedMerge(OrderByDirection.Ascending, sequenceB).Take(2).Consume();
```

Comment:
Would be great if we can update this to bring it in line with the similar test that was added for `Interleave` in PR #696:

https://github.com/morelinq/MoreLINQ/blob/e98d6322c75af1902359947661b85a867c40df50/MoreLinq.Test/InterleaveTest.cs#L94-L106

It just needs the following changes (use testing sequences, dispose them & always end in an assertion):

```suggestion
            using var sequenceA = TestingSequence.Of(1, 3);
            using var sequenceB = MoreEnumerable.From(() => 2, () => throw new TestException())
                                                .AsTestingSequence();

            var result = sequenceA.SortedMerge(OrderByDirection.Ascending, sequenceB).Take(2);

            Assert.That(() => result.Consume(), Throws.Nothing);
```